markdown/events/2021/bytesize-16-module-test-data.md

```diff
@@ -27,3 +28,40 @@ This week, Kevin Menden ([@KevinMenden](http://github.com/KevinMenden/)) will pr
+
+[3:25](https://youtu.be/QXfAerydAT0?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=205) The next thing that comes with this is the `test.yml` file, where you define some information about the test. In this case, the name of the test is `fastqc single-end` i.e. fastqc for single-end data. Then you define the commands for the test, and a tag for the test, which is used to run the rest by GitHub actions. Then also importantly, we have the output files defined. In this case, the module produces both a .html file and a .zip file as outputs, and usually there are also md5 sums in here so that we can see that the output of the module is the same. That’s not the case here, but it doesn’t matter.
+
+[4:14](https://youtu.be/QXfAerydAT0?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=254) So those were the files that made up the tests. Now, you might wonder how you actually know these dictionary keys I showed you earlier. To define the file that we’d like to use for testing, we have this command file which is a Nextflow command to define a file. Then we have the test data parameter, which is basically a dictionary. Within that, we look for the `test_1_fastqc_gz` entry using the `sarscov2` and `illumina` keys. If you’re wondering where to get these keys from..
+
+[4:54](https://youtu.be/QXfAerydAT0?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=294).. For this the most important file is the `test_data.config` file, which is stored in `modules/tests/config/test_data.config directory`. So this file is used to define all these different test data files that are available. It has a link to the test data directory, which is the nf-core test dataset repository, and specifically the modules branch of this repository where we store all the test data. There are links to specific files in this directory, and these links are again linked in this dictionary. This is just a small example of this whole test data parameter dictionary. The reason we do it like this is so that we are able to change these links at some point without changing the code in the `test_main.nf` files of the modules because you will only have these keys here and these entries are defined in your script, so you don’t really need to care about what’s happening on this side of the code. So this is where I can look for the keys and the data I need in the file.
```

Comment:
```suggestion
[4:54](https://youtu.be/QXfAerydAT0?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=294).. For this the most important file is the `test_data.config` file, which is stored in `modules/tests/config/test_data.config directory`. So this file is used to define all these different test data files that are available. It has a link to the test data directory, which is the nf-core test dataset repository, and specifically the modules branch of this repository where we store all the test data. There are links to specific files in this directory, and these links are again linked in this dictionary. This is just a small example of this whole test data parameter dictionary. The reason we do it like this is so that we are able to change these links at some point without changing the code in the `test_main.nf` files of the modules because you will only have these keys here and these entries are defined in your script, so you don’t really need to care about what’s happening on this side of the code. So this is where I can look for the keys and the data I need in the file.
```